src/osd/OSDMap.h

```diff
@@ -552,8 +552,8 @@ class OSDMap {
     CEPH_FEATUREMASK_SERVER_LUMINOUS |
     CEPH_FEATUREMASK_SERVER_MIMIC |
     CEPH_FEATUREMASK_SERVER_NAUTILUS |
-    CEPH_FEATUREMASK_SERVER_OCTOPUS;
-  // TODO: add a SIGNIFICANT_FEATURE for stretch mode...
+    CEPH_FEATUREMASK_SERVER_OCTOPUS |
+    CEPH_FEATUREMASK_STRETCH_MODE;
```

Comment:
I think we need a osdmap flag as well to determine when it's safe to enable stretch mode, so that we can prevent osds without that flag from booting and using an incorrect choose_acting() algorithm. see pg_log_hardlimit for a similar situation - just described for another issue here: https://github.com/ceph/ceph/pull/35938#issuecomment-656183558

Reply:
Hmm I thought adding this to significant features would do that (it will certainly prevent OSDs from decoding the various structs correctly, as when crush peering values are set the encoding compatv == struct_v!).

It looks like the pg_log_hardlimit does something in the OSDMonitor to prevent OSDs booting in at all, but don't we have OSDMap flags somewhere to handle this case based on the required feature set? It looks like maybe adding it to OSDMap::get_features() for the OSD entity type?